src/modules/esl-utils/dom/scroll/parent.ts

```diff
@@ -3,32 +3,36 @@ import {isElement, getNodeName, getParentNode} from '../api';
 /**
  * Get the list of all scroll parents, up the list of ancestors until we get to the top window object.
  * @param element - element for which you want to get the list of all scroll parents
- * @param list - array of elements to concatenate with the list of all scroll parents of element (optional)
+ * @param topElement - an element which element considered a final scrollable parent target (optional, defaults to element.ownerDocument?.body)
  */
-export function getListScrollParents(element: Element, list: Element[] = []): Element[] {
-  const scrollParent = getScrollParent(element);
-  const isBody = scrollParent === element.ownerDocument?.body;
-  const target = isBody
-    ? isScrollable(scrollParent) ? scrollParent : []
-    : scrollParent;
-
-  const updatedList = list.concat(target);
-  return isBody
-    ? updatedList
-    : updatedList.concat(getListScrollParents(getParentNode(scrollParent) as Element));
+export function getListScrollParents(element: Element, topElement?: Element): Element[] {
+  const actualTarget = topElement || element.ownerDocument?.body;
```

Comment:
Is it target? It is still a boundary/terminal element to stop the search (It might be some sort of target in place of usages but it is not according to what the exact method does)